src/misc/mntent.c

```diff
@@ -69,12 +111,41 @@ struct mntent *getmntent(FILE *f)
 	return getmntent_r(f, &mnt, SENTINEL, 0);
 }
 
+static int escape_and_write_string(FILE *f, const char* str)
+{
+	char c;
+	int error_occured = 0;
+	while(str && !error_occured && (c = *str++) != 0) {
+		if(c == '\040') // space
```

Comment:
i'd use a table here too, something like
```c
const char[] chars = "\\\040\011\012";
if((p = strchr(chars, c)))  {
static const char* replacement[] = {"\\", "040", "011", "012"};
fputc('\\');
fputs(replacement[p-chars];
} else fputc(c);
```
untested, but you get the idea.

Reply:
Thanks, will do that next!

Reply:
> i'd use a table here too, something like
> ```c
> const char[] chars = "\\\040\011\012";
> if((p = strchr(chars, c)))  {
> static const char* replacement[] = {"\\", "040", "011", "012"};
> fputc('\\');
> fputs(replacement[p-chars];
> } else fputc(c);
> ```
> untested, but you get the idea.

That seems more complicated and error prone than doing this manually? Especially since it's only four entries.

(I have the same instinct about the other table.)

Glibc does some interesting arithmetic here that we could replicate: they generate the octal escape directly from the char when interpreted in octal.

In any case, if you want a table, I can give you a table.

Reply:
@rofl0r Please see new version.  Instead of the table, I ported some clever arithmetic.  Hope you like it?